### ISSUE TITLE
Issue / Question - Unable to Add Item to Sale after CSV Import

### ISSUE THREAD

**Issue by marcvasquez** [jekkos/opensourcepos#1558]
### Issue / Question / Bug

Before submitting an issue please make sure you tick (add an x between the square brackets with no spaces) the following check boxes:

- [x] I'm reporting an issue of an unmodified OSPOS installation
- [x] I checked open and closed issues database and no similar issue was already discussed (please make sure you searched!)
- [x] I read the README, WHATS_NEW and UPGRADE
- [x] I read the FAQ (https://github.com/jekkos/opensourcepos#faq) for any known install and/or upgrade gotchas (in specific PHP has php-gd, php-intl, sockets and etc. installed)
- [x] I read the wiki
- [x] I ran any database upgrade scripts (e.g. database/2.4_to_3.0.sql), and migrating function
- [x] I'm aware the latest master could be a development version and therefore not stable
- [x] I know the version of OSPOS and git commit hash (check the footer of your OSPOS), the name and version of OS, Web server, PHP and MySQL and will add them to my issue report

### Installation information

- OSPOS version is:  3.1.0
- OSPOS git commit hash is:  4f5ad57. 
- PHP version is:   (e.g. 5.5, 5.6, 7.0, 7.1) Used Docker Container
- MySQL or MariaDB version is:   (e.g. MySQL 5.5, MySQL 5.6, MySQL 5.7, MariaDB 10.0, MariaDB 10.1, MariaDB 10.2)  Docker Container
- OS and version is:   (e.g. CentOS 6.9, Ubuntu 16.4, Windows 10) Ubuntu 16.04.2 LTS
- WebServer is:   (e.g. Apache 2.2, Apache 2.4, Nginx 1.12, Nginx 1.13) Docker (Apache 2.4)
- (If applicable) Installation package for the LAMP/LEMP stack is:   (e.g. WAMP, XAMPP)

### Expected behaviour

Items add to cart correctly.

### Actual behaviour

Trying to add items to a sale, getting red bar across top with error "Unable to add item to sale". Only happens with items imported via CSV

### Steps to reproduce the issue

I imported items via CSV.  Items show up correctly in the listing of items, and are editable via the table, but when I try to add them to a sale it gives the error message.  Manually created items are working correctly in the items table and the sale page, but anything from CSV import will not add to a sale. Not sure if I did anything wrong with them, didn't get any errors when importing. Running the docker version locally on an Ubuntu VM. Thanks in advance.

![fjx](https://user-images.githubusercontent.com/31709573/30133589-6e5a3680-9322-11e7-8d8e-fe1911e9307b.png) Item in list
![fjx in dropdown](https://user-images.githubusercontent.com/31709573/30133590-6e5c9902-9322-11e7-8f88-5b910b6fcf0a.png) Item in Dropdown
![sale1](https://user-images.githubusercontent.com/31709573/30133591-6e5f9f62-9322-11e7-9f2a-ef9c372d93df.png) Error after selecting item

**Comment by daN4cat** [comment]
@marcvasquez could you please share one csv file example that you know doesn't work?
Thanks in advance for the help.

**Comment by daN4cat** [comment]
I need a means to securely reproduce the bug in order to investigate and fix.

**Comment by marcvasquez** [comment]
Sure. Should I attach here or do you prefer a dropbox or Google drive link?

**Comment by daN4cat** [comment]
Please attach here. Thanks

**Comment by marcvasquez** [comment]
[Sixth.xlsx](https://github.com/jekkos/opensourcepos/files/1282686/Sixth.xlsx)
It won't upload as CSV but that's one of the workbooks and I just saved as CSV then uploaded. Thanks!

**Comment by marcvasquez** [comment]
As an interesting side test, I am able to receive inventory against the part #s that were imported, and when I go and view an item after receiving it does show the new qty, but still is unable to add to a sale.
Not sure if that helps at all

**Comment by marcvasquez** [comment]
Further testing - I tried creating a new item from the New Item field and that too is not allowing me to add it to the sale.  Only items I manually created before the import work, now nothing import or new is working in the sale. Thanks!

**Comment by marcvasquez** [comment]
Further investigation - I think one of the file imports is causing the issue.  Started a new Ubuntu VM, installed docker and launched.  I have a total of 6 csv files.  Imported 3 of them, testing a sale item after each, found that after Third.csv I am no longer able to add items to sales, including previously imported and successful items.
[Third.xlsx](https://github.com/jekkos/opensourcepos/files/1285200/Third.xlsx)

**Comment by marcvasquez** [comment]
Sorry for so many posts but I am trying to do as much trouble shooting as possible.. looks like files 1 and 3 are my culprits.  Was able to import the other 4 without a problem, but once either one of these is loaded it causes the sale function to break. Any help is greatly appreciated. Here is the other file in question.

[First.xlsx](https://github.com/jekkos/opensourcepos/files/1285256/First.xlsx)

**Comment by daN4cat** [comment]
@marcvasquez probably you need to set a location in your cvs file, that could be the root cause of your issue.

**Comment by marcvasquez** [comment]
Thank you - will try and see what happens!

**Comment by daN4cat** [comment]
@marcvasquez can you please give me an update? I believe the issue was the missing location.

**Comment by bkrajendra** [comment]
I'm also trying to import from CSV total 1090 items, I split CSV into 3 parts file1.csv, file2.csv file3.csv.
When import, All items get imported without any issue. But Some of items get added to sale but some of them could not get added to sale.
I've tried to see all possibilities about type of text in items, but there are some items which have longs names, special characters and + signs get added perfectly.

The behaviour is totally random. Out of all data there are some item that works ok with sale but some does not.

Ive also added location to 1, i hope its correct. 
@marcvasquez  is your issue resolve? if yes please let me know as I need urgent help.
@daN4cat Any idea what could be causing this.

Ive tried using latest release. Also tried with latest master today. But issue is same with both.

**Comment by bkrajendra** [comment]
do i need to create new issue, as this seems to be closed.?

**Comment by daN4cat** [comment]
How can we test this?

**Comment by daN4cat** [comment]
I mean reproduce the issue?

**Comment by bkrajendra** [comment]
is there any other way i can give u csv file?

**Comment by bkrajendra** [comment]
ok! now i tried with entire 1090 items and it got imported in one go. I've set location to 1 and supplier id to 2. I also made it it sure to have supplier with id 2. Also one of my products was missing supplier id. I corrected it. 
Now what i observer is items are not listed in Items section. But everything else is working fine.. Receivings, sales and all working ok.

Items get displayed in Item section only after sale. So initially i used all quantity =0 while importing, so i thought it might be due to that. But then just now I tried by setting quantity =1 ... but still problem is same.
No item is displayed in Items section until we sell it . Is it new functionality or something... looks weired!

**Comment by bkrajendra** [comment]
All items are back after I logged out and logged in again. It took little time to login, it was like something was going on at background. And now all Items are showing up in Items section.
rest all systems are working fine.
Will test tomorrow on production system and report back here.
Thanks all!

**Comment by daN4cat** [comment]
If you could not see the items then it's a problem with the transaction table. It need to contain the date and time of the csv upload. But that should go wrong always if it's not working. I don't understand what your database is doing to say they appear later. That doesn't make sense to me. Let me know how it goes.

**Comment by daN4cat** [comment]
Please check one item with the inventory button on the right and see if the log reports the csv import as first line.

**Comment by ashishkaushik** [comment]
I am facing this issue with any item added manually via New Item Button. 
Any pointers ?
